tests/utils.go

```diff
@@ -1950,32 +1950,66 @@ func createNamespaces() {
 }
 
 func NewRandomDataVolumeWithRegistryImport(imageUrl, namespace string, accessMode k8sv1.PersistentVolumeAccessMode) *cdiv1.DataVolume {
-	return NewRandomDataVolumeWithRegistryImportInStorageClass(imageUrl, namespace, Config.StorageClassLocal, accessMode)
+	sc, exists := GetRWOFileSystemStorageClass()
+	if accessMode == k8sv1.ReadWriteMany {
+		sc, exists = GetRWXFileSystemStorageClass()
+	}
+	if !exists {
+		Skip("Skip test when Filesystem storage is not present")
+	}
+	return NewRandomDataVolumeWithRegistryImportInStorageClass(imageUrl, namespace, sc, accessMode)
 }
 
 func NewRandomBlankDataVolume(namespace, storageClass, size string, accessMode k8sv1.PersistentVolumeAccessMode, volumeMode k8sv1.PersistentVolumeMode) *cdiv1.DataVolume {
 	return newRandomBlankDataVolume(namespace, storageClass, size, accessMode, volumeMode)
 }
 
-func NewRandomVirtualMachineInstanceWithOCSDisk(imageUrl, namespace string, accessMode k8sv1.PersistentVolumeAccessMode, volMode k8sv1.PersistentVolumeMode) (*v1.VirtualMachineInstance, *cdiv1.DataVolume) {
-	if !HasCDI() {
-		Skip("Skip DataVolume tests when CDI is not present")
-	}
-	sc, exists := GetCephStorageClass()
-	if !exists {
-		Skip("Skip OCS tests when Ceph is not present")
-	}
+func newRandomVirtualMachineInstanceWithDisk(imageUrl, namespace, sc string, accessMode k8sv1.PersistentVolumeAccessMode, volMode k8sv1.PersistentVolumeMode) (*v1.VirtualMachineInstance, *cdiv1.DataVolume) {
 	virtCli, err := kubecli.GetKubevirtClient()
 	util2.PanicOnError(err)
 
 	dv := NewRandomDataVolumeWithRegistryImportInStorageClass(imageUrl, namespace, sc, accessMode)
 	dv.Spec.PVC.VolumeMode = &volMode
 	_, err = virtCli.CdiClient().CdiV1beta1().DataVolumes(dv.Namespace).Create(context.Background(), dv, metav1.CreateOptions{})
 	Expect(err).ToNot(HaveOccurred())
-	Eventually(ThisDV(dv), 240).Should(HaveSucceeded())
+	if HasBindingModeWaitForFirstConsumer() {
+		Eventually(ThisDV(dv), 30).Should(BeInPhase(cdiv1.WaitForFirstConsumer))
+	} else {
+		Eventually(ThisDV(dv), 240).Should(HaveSucceeded())
+	}
 	return NewRandomVMIWithDataVolume(dv.Name), dv
 }
 
+func NewRandomVirtualMachineInstanceWithFileDisk(imageUrl, namespace string, accessMode k8sv1.PersistentVolumeAccessMode) (*v1.VirtualMachineInstance, *cdiv1.DataVolume) {
```

Comment:
I am trying to decide if something like this is more DRY or not. I am on the fence. There is just a lot of duplication between these two functions.
```go
func NewRandomVirtualMachineInstanceWithFileDisk(imageUrl, namespace string, accessMode k8sv1.PersistentVolumeAccessMode) (*v1.VirtualMachineInstance, *cdiv1.DataVolume) {
  return NewRandomVirtualMachineInstanceWithDisk(imageUrl, namespace, accessMode, k8sv1.PersistentVolumeFilesystem, getFileSystemStorageClassByAccessMode)
}

func NewRandomVirtualMachineInstanceWithBlockDisk(imageUrl, namespace string, accessMode k8sv1.PersistentVolumeAccessMode) (*v1.VirtualMachineInstance, *cdiv1.DataVolume) {
  return NewRandomVirtualMachineInstanceWithDisk(imageUrl, namespace, accessMode, k8sv1.PersistentVolumeBlock, getBlockStorageClassByAccessMode)
}

func NewRandomVirtualMachineInstanceWithDisk(imageUrl, namespace string, accessMode k8sv1.PersistentVolumeAccessMode, volumeMode *k8sv1.PersistentVolumeMode, getSc func(volumeMode *k8sv1.PersistentVolumeAccessMode) *storagev1.StorageClass)  (*v1.VirtualMachineInstance, *cdiv1.DataVolume) {
    if !HasCDI() {
        Skip("Skip DataVolume tests when CDI is not present")
    }

    return newRandomVirtualMachineInstanceWithDisk(imageUrl, namespace, getSc(accessMode), accessMode, volumeMode)
}

func getFileSystemStorageClassByAccessMode(accessMode k8sv1.PersistentVolumeAccessMode) *storagev1.StorageClass {
    sc, exists := GetRWOFileSystemStorageClass()
    if accessMode == k8sv1.ReadWriteMany {
        sc, exists = GetRWXFileSystemStorageClass()
    }
    if !exists {
        Skip("Skip test when Filesystem storage is not present")
    }
}

func getBlockStorageClassByAccessMode(accessMode k8sv1.PersistentVolumeAccessMode) *storagev1.StorageClass {
    sc, exists := GetRWOBlockStorageClass()
    if accessMode == k8sv1.ReadWriteMany {
        sc, exists = GetRWXBlockStorageClass()
    }
    if !exists {
        Skip("Skip test when Block storage is not present")
    }
}
```

Reply:
I think this only removes the duplication of the check for had CDI and moved the other code to another 2 functions.. seems a bit of more complication and jumping through several functions just for that..

Reply:
I am not particularly passionate about either one. Other people might be though.